--- origsrc/graphviz-2.18/Doxyfile.in	2007-08-31 18:12:40.000000000 -0500
+++ src/graphviz-2.18/Doxyfile.in	2008-03-20 14:32:27.265625000 -0500
@@ -462,23 +462,23 @@
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  = cmd/dot \
-                         lib/cdt \
-                         lib/circogen \
-                         lib/common \
-                         lib/dotgen \
-                         lib/fdpgen \
-                         lib/graph \
-                         lib/gvc \
-                         lib/neatogen \
-                         lib/pack \
-                         lib/pathplan \
-                         lib/twopigen \
-                         plugin/core \
-                         plugin/pango \
-                         plugin/dot_layout \
-                         plugin/neato_layout \
-                         tclpkg/gv
+INPUT                  = @top_srcdir@/cmd/dot \
+                         @top_srcdir@/lib/cdt \
+                         @top_srcdir@/lib/circogen \
+                         @top_srcdir@/lib/common \
+                         @top_srcdir@/lib/dotgen \
+                         @top_srcdir@/lib/fdpgen \
+                         @top_srcdir@/lib/graph \
+                         @top_srcdir@/lib/gvc \
+                         @top_srcdir@/lib/neatogen \
+                         @top_srcdir@/lib/pack \
+                         @top_srcdir@/lib/pathplan \
+                         @top_srcdir@/lib/twopigen \
+                         @top_srcdir@/plugin/core \
+                         @top_srcdir@/plugin/pango \
+                         @top_srcdir@/plugin/dot_layout \
+                         @top_srcdir@/plugin/neato_layout \
+                         @top_srcdir@/tclpkg/gv
 
 # This tag can be used to specify the character encoding of the source files that 
 # doxygen parses. Internally doxygen uses the UTF-8 encoding, which is also the default 
