--- origsrc/graphviz-2.24.0/configure.ac	2009-11-04 20:21:02.716175800 -0600
+++ src/graphviz-2.24.0/configure.ac	2009-11-04 20:20:31.766133100 -0600
@@ -83,7 +83,7 @@ AC_SUBST(GVPLUGIN_VERSION)
 GRAPHVIZ_VERSION_BUILD=0
 GRAPHVIZ_VERSION_REVISION=0
 case "${host_os}" in
-  *uwin* | *cygwin* | *mingw32* )
+  *uwin* | *mingw32* )
 	# Windows version numbers
 	# For the build number: months since Jan 2000, day of month from the timestamp
 	# For the revision number: hour, minute from the timestamp
@@ -659,7 +659,7 @@ else
 	  use_guile="No (guile is too old)"
         else
           GUILE_INCLUDES=
-          GUILE_LIBS=
+          GUILE_LIBS=-lguile
           AC_CHECK_HEADER(libguile.h,,[
             use_guile="No (guile header missing)"
             GUILE=
@@ -1017,7 +1017,7 @@ else
         fi
         PHP_INSTALL_DIR="/usr/lib${LIBPOSTFIX}/php/modules"
         PHP_INSTALL_DATADIR="/usr/share/php"
-        PHP_LIBS=
+        PHP_LIBS=-lphp5lib
         save_CPPFLAGS=$CPPFLAGS
         CPPFLAGS="$CPPFLAGS $PHP_INCLUDES"
         AC_CHECK_HEADER(php.h,,[
@@ -1498,7 +1498,6 @@ if test "x$use_tcl" = "x"; then
   # TCL STUBS support is required
   if test "x${TCL_SUPPORTS_STUBS}" = "x1"; then
     use_tcl="Yes"
-    TCL_CFLAGS="${TCL_CFLAGS} -DUSE_TCL_STUBS"
     AC_DEFINE_UNQUOTED(HAVE_TCL,1,
 	[Define if you have the tcl library])
   else
@@ -1550,7 +1549,7 @@ AC_SUBST(TCL_SHLIB_SUFFIX)
 #AC_SUBST(TCL_RANLIB)
 #AC_SUBST(TCL_LIB_FLAG)
 #AC_SUBST(TCL_BUILD_LIB_SPEC)
-#AC_SUBST(TCL_LIB_SPEC)
+AC_SUBST(TCL_LIB_SPEC)
 #AC_SUBST(TCL_LIB_VERSIONS_OK)
 #AC_SUBST(TCL_SHARED_LIB_SUFFIX)
 #AC_SUBST(TCL_UNSHARED_LIB_SUFFIX)
@@ -1649,10 +1648,6 @@ else
 fi
 AM_CONDITIONAL(WITH_TK, [test "x$use_tk" = "xYes"])
 
-# NB. No TK_SUPPORTS_STUBS set by tkConfig.sh
-if test "${TCL_SUPPORTS_STUBS}" = "1"; then
-    TK_CFLAGS="${TK_CFLAGS} -DUSE_TK_STUBS"
-fi
 
 AC_SUBST(TK_VERSION)
 AC_SUBST(TK_MAJOR_VERSION)
--- origsrc/graphviz-2.24.0/tclpkg/gdtclft/Makefile.am	2009-11-05 01:05:00.623075200 -0600
+++ src/graphviz-2.24.0/tclpkg/gdtclft/Makefile.am	2009-11-05 15:50:29.174321600 -0600
@@ -11,8 +11,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/tclpkg/tclhandle \
 	@GD_INCLUDES@ @TCL_INCLUDES@
 
-LIBS = @LIBS@ -lc
-
 AM_CFLAGS = @TCL_CFLAGS@
 
 man = gdtclft.3tcl
@@ -28,7 +28,7 @@ endif
 
 libgdtclft_la_SOURCES = gdtclft.c
 
-libgdtclft_la_LDFLAGS = -no-undefined
+libgdtclft_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if WITH_LIBGD
 if WITH_MYLIBGD
--- origsrc/graphviz-2.24.0/tclpkg/gv/Makefile.am	2009-06-11 11:03:37.000000000 -0500
+++ src/graphviz-2.24.0/tclpkg/gv/Makefile.am	2009-11-04 20:19:09.536019600 -0600
@@ -20,8 +20,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/cdt \
 	-I$(top_srcdir)/lib/pathplan
 
-LIBS = -lc
-
 BASESOURCES = gv.cpp gv.i gv_builtins.c
 BASELIBS = $(top_builddir)/lib/gvc/libgvc.la \
 	$(top_builddir)/lib/$(GRAPH)/lib$(GRAPH).la \
--- origsrc/graphviz-2.24.0/tclpkg/tcldot/Makefile.am	2009-06-02 20:10:58.000000000 -0500
+++ src/graphviz-2.24.0/tclpkg/tcldot/Makefile.am	2009-11-04 20:19:09.536019600 -0600
@@ -23,8 +23,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/tclpkg/tclhandle $(GD_INCLUDES) $(TCL_INCLUDES)
 
-LIBS = -lc
-
 AM_CFLAGS = $(TCL_CFLAGS)
 
 man = tcldot.3tcl
@@ -45,7 +43,7 @@ endif
 
 libtcldot_la_SOURCES = tcldot.c no_builtins.c $(CODEGENS) $(GDTCLFT)
 libtcldot_la_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
-libtcldot_la_LDFLAGS = -no-undefined 
+libtcldot_la_LDFLAGS = -module -avoid-version -no-undefined 
 
 libtcldot_la_LIBADD = $(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
 	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
@@ -67,7 +65,7 @@ libtcldot_la_LIBADD += $(LIBGEN_LIBS) $(
 
 libtcldot_builtin_la_SOURCES = tcldot.c tcldot_builtins.c $(CODEGENS) $(GDTCLFT)
 libtcldot_builtin_la_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
-libtcldot_builtin_la_LDFLAGS =
+libtcldot_builtin_la_LDFLAGS = -module -avoid-version -no-undefined
 
 libtcldot_builtin_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
--- origsrc/graphviz-2.24.0/tclpkg/tclpathplan/Makefile.am	2009-06-02 20:10:58.000000000 -0500
+++ src/graphviz-2.24.0/tclpkg/tclpathplan/Makefile.am	2009-11-04 20:19:09.536019600 -0600
@@ -18,7 +18,7 @@ pkgtcl_LTLIBRARIES = libtclplan.la
 endif
 
 libtclplan_la_SOURCES = find_ints.c intersect.c makecw.c tclpathplan.c wrapper.c
-libtclplan_la_LDFLAGS = -no-undefined
+libtclplan_la_LDFLAGS = -module -avoid-version -no-undefined
 libtclplan_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
 	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
--- origsrc/graphviz-2.24.0/tclpkg/tclstubs/Makefile.am	2009-06-02 20:10:58.000000000 -0500
+++ src/graphviz-2.24.0/tclpkg/tclstubs/Makefile.am	2009-11-04 20:19:09.546019600 -0600
@@ -3,7 +3,7 @@
 
 AM_CPPFLAGS = @TCL_CFLAGS@ @TCL_INCLUDES@
 
-LIBS = @TCL_STUB_LIB_SPEC@
+LIBS = @TCL_LIB_SPEC@
 
 if WITH_TCL
 noinst_LTLIBRARIES = libtclstubs_C.la
--- origsrc/graphviz-2.24.0/tclpkg/tkspline/Makefile.am	2009-06-02 20:10:58.000000000 -0500
+++ src/graphviz-2.24.0/tclpkg/tkspline/Makefile.am	2009-11-04 20:19:09.546019600 -0600
@@ -26,7 +26,7 @@ man_MANS = $(man)
 pdf_DATA = $(pdf)
 endif
 
-libtkspline_la_LDFLAGS = -no-undefined 
+libtkspline_la_LDFLAGS = -module -avoid-version -no-undefined 
 libtkspline_la_SOURCES = tkspline.c
 libtkspline_la_LIBADD = \
 	$(top_builddir)/tclpkg/tkstubs/libtkstubs_C.la \
--- origsrc/graphviz-2.24.0/tclpkg/tkstubs/Makefile.am	2009-06-02 20:10:58.000000000 -0500
+++ src/graphviz-2.24.0/tclpkg/tkstubs/Makefile.am	2009-11-04 20:19:09.556019600 -0600
@@ -3,7 +3,7 @@
 
 AM_CPPFLAGS = @TCL_CFLAGS@ @TK_CFLAGS@ @TCL_INCLUDES@  @TK_INCLUDES@ @TK_XINCLUDES@
                                                                                 
-LIBS = @TK_STUB_LIB_SPEC@
+LIBS = @TK_LIB_SPEC@
 
 if WITH_TK
 noinst_LTLIBRARIES = libtkstubs_C.la
