--- origsrc/graphviz-2.26.3/configure.ac	2010-07-02 15:28:45.608113800 -0500
+++ src/graphviz-2.26.3/configure.ac	2010-06-23 03:56:09.000000000 -0500
@@ -83,7 +83,7 @@ AC_SUBST(GVPLUGIN_VERSION)
 GRAPHVIZ_VERSION_BUILD=0
 GRAPHVIZ_VERSION_REVISION=0
 case "${host_os}" in
-  *uwin* | *cygwin* | *mingw32* )
+  *uwin* | *mingw32* )
 	# Windows version numbers
 	# For the build number: months since Jan 2000, day of month from the timestamp
 	# For the revision number: hour, minute from the timestamp
@@ -649,7 +649,7 @@ else
 	  use_guile="No (guile is too old)"
         else
           GUILE_INCLUDES=
-          GUILE_LIBS=
+          GUILE_LIBS=-lguile
           AC_CHECK_HEADER(libguile.h,,[
             use_guile="No (guile header missing)"
             GUILE=
@@ -1014,7 +1014,7 @@ else
         fi
         PHP_INSTALL_DIR="/usr/lib${LIBPOSTFIX}/php/modules"
         PHP_INSTALL_DATADIR="/usr/share/php"
-        PHP_LIBS=
+        PHP_LIBS=-lphp5lib
         save_CPPFLAGS=$CPPFLAGS
         CPPFLAGS="$CPPFLAGS $PHP_INCLUDES"
         AC_CHECK_HEADER(php.h,,[
--- origsrc/graphviz-2.26.3/config/config_perl.pl	2008-07-14 07:54:04.000000000 -0500
+++ src/graphviz-2.26.3/config/config_perl.pl	2010-07-05 17:45:20.489567700 -0500
@@ -4,7 +4,7 @@ use Config;
 	$archlib = $Config{archlib};
 	$libperl = $Config{libperl};
 	$libperl =~ s/lib([^\.]+).*/$1/;
-	print "-L$archlib/CORE -l$libperl";
+	print "-L$archlib/CORE -lperl";
 }
 if ($ARGV[0] eq "PERL_INCLUDES") {
 	$archlib = $Config{archlib};
--- origsrc/graphviz-2.26.3/tclpkg/Makefile.am	2009-06-02 20:10:58.000000000 -0500
+++ src/graphviz-2.26.3/tclpkg/Makefile.am	2010-07-02 15:18:27.527761700 -0500
@@ -22,7 +22,7 @@ if WITH_TCL
 pkgindex_DATA = pkgIndex.tcl
 endif
 
-SUBDIRS = tclstubs tkstubs tclhandle gdtclft tcldot tclpathplan tkspline gv
+SUBDIRS = tclhandle gdtclft tcldot tclpathplan tkspline gv
 
 pkgIndex.tcl: @TCL_PKGINDEX@ @TCL_PKGINDEX_GD@ @TCL_PKGINDEX_SWIG@ @TK_PKGINDEX@
 	echo "# end" >end
--- origsrc/graphviz-2.26.3/tclpkg/gdtclft/Makefile.am	2010-06-23 03:56:09.928567500 -0500
+++ src/graphviz-2.26.3/tclpkg/gdtclft/Makefile.am	2010-06-23 04:22:08.898735500 -0500
@@ -11,8 +11,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/tclpkg/tclhandle \
 	@GD_INCLUDES@ @TCL_INCLUDES@
 
-LIBS = @LIBS@ -lc
-
 AM_CFLAGS = @TCL_CFLAGS@
 
 man = gdtclft.3tcl
@@ -28,19 +26,19 @@ endif
 
 libgdtclft_la_SOURCES = gdtclft.c
 
-libgdtclft_la_LDFLAGS = -no-undefined
+libgdtclft_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if WITH_LIBGD
 if WITH_MYLIBGD
 libgdtclft_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	$(top_builddir)/lib/gd/libgvgd_C.la \
 	@ICONV_LIBS@ @FONTCONFIG_LIBS@ @FREETYPE2_LIBS@ @JPEG_LIBS@ @PNG_LIBS@ @Z_LIBS@ @MATH_LIBS@
 else
 libgdtclft_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	@GD_LIBS@ @ICONV_LIBS@ @FONTCONFIG_LIBS@ @FREETYPE2_LIBS@ @JPEG_LIBS@ @PNG_LIBS@ @Z_LIBS@ @MATH_LIBS@
 endif
 
--- origsrc/graphviz-2.26.3/tclpkg/gv/Makefile.am	2010-06-23 03:56:09.932567800 -0500
+++ src/graphviz-2.26.3/tclpkg/gv/Makefile.am	2010-07-02 15:21:17.046457600 -0500
@@ -20,8 +20,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/cdt \
 	-I$(top_srcdir)/lib/pathplan
 
-LIBS = -lc
-
 BASESOURCES = gv.cpp gv.i gv_builtins.c
 BASELIBS = $(top_builddir)/lib/gvc/libgvc.la \
 	$(top_builddir)/lib/$(GRAPH)/lib$(GRAPH).la \
@@ -224,9 +222,9 @@ pkgtcldir = $(pkglibdir)/tcl
 TCL_data = 
 nodist_libgv_tcl_la_SOURCES = gv_tcl.cpp $(TCL_data)
 libgv_tcl_la_SOURCES = $(BASESOURCES) gv_tcl_init.c
-libgv_tcl_la_LIBADD = $(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la $(BASELIBS) $(TCL_LIBS)
+libgv_tcl_la_LIBADD = $(TCL_STUB_LIB_SPEC) $(BASELIBS) $(TCL_LIBS)
 libgv_tcl_la_LDFLAGS = -module -avoid-version
-libgv_tcl_la_CPPFLAGS = $(BASECPPFLAGS) $(TCL_INCLUDES)
+libgv_tcl_la_CPPFLAGS = $(BASECPPFLAGS) $(TCL_CFLAGS) $(TCL_INCLUDES)
 $(TCL_data): gv_tcl.cpp
 gv_tcl.cpp: gv.i
 	$(SWIG) -c++ -tcl -namespace -o gv_tcl.cpp $(srcdir)/gv.i
--- origsrc/graphviz-2.26.3/tclpkg/tcldot/Makefile.am	2010-06-23 03:56:09.937568100 -0500
+++ src/graphviz-2.26.3/tclpkg/tcldot/Makefile.am	2010-06-23 04:21:51.457738000 -0500
@@ -23,8 +23,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/tclpkg/tclhandle $(GD_INCLUDES) $(TCL_INCLUDES)
 
-LIBS = -lc
-
 AM_CFLAGS = $(TCL_CFLAGS)
 
 man = tcldot.3tcl
@@ -45,10 +43,10 @@ endif
 
 libtcldot_la_SOURCES = tcldot.c no_builtins.c $(GDTCLFT)
 libtcldot_la_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
-libtcldot_la_LDFLAGS = -no-undefined 
+libtcldot_la_LDFLAGS = -module -avoid-version -no-undefined 
 
 libtcldot_la_LIBADD = $(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	$(top_builddir)/lib/gvc/libgvc.la 
 
 libtcldot_la_DEPENDENCIES =  $(top_builddir)/lib/gvc/libgvc.la
@@ -67,7 +65,7 @@ libtcldot_la_LIBADD += $(LIBGEN_LIBS) $(
 
 libtcldot_builtin_la_SOURCES = tcldot.c tcldot_builtins.c $(GDTCLFT)
 libtcldot_builtin_la_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
-libtcldot_builtin_la_LDFLAGS =
+libtcldot_builtin_la_LDFLAGS = -module -avoid-version -no-undefined
 
 libtcldot_builtin_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
@@ -78,7 +77,7 @@	$(top_builddir)/lib/gvc/libgvc_C.la \
 	$(top_builddir)/lib/$(GRAPH)/lib$(GRAPH)_C.la \
 	$(top_builddir)/lib/cdt/libcdt_C.la \
 	$(top_builddir)/lib/pathplan/libpathplan_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la $(GTS_LIBS)
+	$(TCL_STUB_LIB_SPEC) $(GTS_LIBS)
 
 if WITH_PANGOCAIRO
 libtcldot_builtin_la_LIBADD += $(top_builddir)/plugin/pango/libgvplugin_pango_C.la $(PANGOCAIRO_LIBS)
--- origsrc/graphviz-2.26.3/tclpkg/tclpathplan/Makefile.am	2010-06-23 03:56:09.943568400 -0500
+++ src/graphviz-2.26.3/tclpkg/tclpathplan/Makefile.am	2010-06-23 04:26:43.451439100 -0500
@@ -18,10 +18,10 @@ pkgtcl_LTLIBRARIES = libtclplan.la
 endif
 
 libtclplan_la_SOURCES = find_ints.c intersect.c makecw.c tclpathplan.c wrapper.c
-libtclplan_la_LDFLAGS = -no-undefined
+libtclplan_la_LDFLAGS = -module -avoid-version -no-undefined
 libtclplan_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	$(top_builddir)/lib/pathplan/libpathplan.la
 
 if WITH_TCL
--- origsrc/graphviz-2.26.3/tclpkg/tkspline/Makefile.am	2010-06-23 03:56:09.947568600 -0500
+++ src/graphviz-2.26.3/tclpkg/tkspline/Makefile.am	2010-07-02 15:18:09.639738500 -0500
@@ -26,11 +26,11 @@ man_MANS = $(man)
 pdf_DATA = $(pdf)
 endif
 
-libtkspline_la_LDFLAGS = -no-undefined 
+libtkspline_la_LDFLAGS = -module -avoid-version -no-undefined 
 libtkspline_la_SOURCES = tkspline.c
 libtkspline_la_LIBADD = \
-	$(top_builddir)/tclpkg/tkstubs/libtkstubs_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la @TK_LIB_SPEC@
+	$(TK_STUB_LIB_SPEC) \
+	$(TCL_STUB_LIB_SPEC)	
 
 if WITH_TK
 all-local: pkgIndex.tcl
