--- origsrc/graphviz-2.28.0/plugin/ming/Makefile.am	2009-06-02 20:10:57.000000000 -0500
+++ src/graphviz-2.28.0/plugin/ming/Makefile.am	2012-07-10 22:48:32.137131800 -0500
@@ -17,7 +17,8 @@ lib_LTLIBRARIES = libgvplugin_ming.la
 else
 pkglib_LTLIBRARIES = libgvplugin_ming.la
 endif
-pkglib_DATA = Bitstream_Vera_Sans.fdb  Bitstream_Vera_Serif.fdb
+fontdir = $(pkglibdir)
+font_DATA = Bitstream_Vera_Sans.fdb  Bitstream_Vera_Serif.fdb
 endif
 
 libgvplugin_ming_C_la_SOURCES = \
--- origsrc/graphviz-2.28.0/tclpkg/gv/Makefile.am	2012-07-10 22:33:27.937499700 -0500
+++ src/graphviz-2.28.0/tclpkg/gv/Makefile.am	2012-07-10 22:42:16.515628200 -0500
@@ -255,7 +255,7 @@ $(TCL_data): gv_tcl.cpp
 gv_tcl.cpp: gv.i
 	$(SWIG) -c++ -tcl -namespace -o gv_tcl.cpp $(srcdir)/gv.i
 
-pkginclude_DATA = gv.i gv.cpp
+pkginclude_HEADERS = gv.i gv.cpp
 pkglib_LTLIBRARIES = 
 man_MANS =
 pdf_DATA = 
