--- origsrc/graphviz-2.36.0/config/config_perl.pl	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/config/config_perl.pl	2014-02-09 21:01:11.880003900 -0600
@@ -4,7 +4,7 @@ if ($ARGV[0] eq "PERL_LIBS") {
 	$archlib = $Config{archlib};
 	$libperl = $Config{libperl};
 	$libperl =~ s/lib([^\.]+).*/$1/;
-	print "-L$archlib/CORE -l$libperl";
+	print "-L$archlib/CORE -lperl -lssp";
 }
 if ($ARGV[0] eq "PERL_INCLUDES") {
 	$archlib = $Config{archlib};
--- origsrc/graphviz-2.36.0/config/config_ruby.rb	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/config/config_ruby.rb	2014-02-09 20:59:34.728447200 -0600
@@ -4,8 +4,10 @@ CONFIG = RbConfig::MAKEFILE_CONFIG
 
 
 case ARGV[0]
-when "archdir"
-    puts RbConfig::expand(CONFIG["archdir"])
+when "rubyhdrdir"
+    puts RbConfig::expand(CONFIG["rubyhdrdir"]) || RbConfig::expand(CONFIG["archdir"])
+when "arch"
+    puts RbConfig::expand(CONFIG["arch"])
 when "lib"
     puts RbConfig::expand(CONFIG["libdir"])
 when "vendorarchdir"
--- origsrc/graphviz-2.36.0/configure.ac	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/configure.ac	2014-02-09 21:01:11.886004300 -0600
@@ -88,7 +88,7 @@ AC_SUBST([GVPLUGIN_VERSION])
 GRAPHVIZ_VERSION_BUILD=0
 GRAPHVIZ_VERSION_REVISION=0
 case "${host_os}" in
-  *uwin* | *cygwin* | *mingw32* )
+  *uwin* | *mingw32* )
 	# Windows version numbers
 	# For the build number: months since Jan 2000, day of month from the timestamp
 	# For the revision number: hour, minute from the timestamp
@@ -1107,7 +1107,7 @@ else
         fi
         PHP_INSTALL_DIR="/usr/lib${LIBPOSTFIX}/php/modules"
         PHP_INSTALL_DATADIR="/usr/share/php"
-        PHP_LIBS=
+        PHP_LIBS=-lphp5
         save_CPPFLAGS=$CPPFLAGS
         CPPFLAGS="$CPPFLAGS $PHP_INCLUDES"
         AC_CHECK_HEADER(php.h,,[
--- origsrc/graphviz-2.36.0/tclpkg/Makefile.am	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/tclpkg/Makefile.am	2014-02-09 21:01:11.890004500 -0600
@@ -25,7 +25,7 @@ if WITH_TCL
 pkgindex_DATA = pkgIndex.tcl
 endif
 
-SUBDIRS = tclstubs tkstubs tclhandle gdtclft tcldot tclpathplan tkspline gv
+SUBDIRS = tclhandle gdtclft tcldot tclpathplan tkspline gv
 
 pkgIndex.tcl: @TCL_PKGINDEX@ @TCL_PKGINDEX_GD@ @TCL_PKGINDEX_SWIG@ @TK_PKGINDEX@
 	echo "# end" >end
@@ -36,7 +36,7 @@ pkgIndex.tcl: @TCL_PKGINDEX@ @TCL_PKGIND
 # separate step, with soft failures, so that a developer can do:
 #    ./configure --prefix=$HOME/graphviz; make; make install
 # without root priviledges.
-install-data-hook:
+install-bindings:
 if WITH_LUA
 	-mkdir -p $(DESTDIR)@LUA_INSTALL_DIR@;
 	if test -w $(DESTDIR)@LUA_INSTALL_DIR@; then \
--- origsrc/graphviz-2.36.0/tclpkg/gdtclft/Makefile.am	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/tclpkg/gdtclft/Makefile.am	2014-02-09 21:01:11.892004600 -0600
@@ -11,8 +11,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/tclpkg/tclhandle \
 	@GD_INCLUDES@ @TCL_INCLUDES@
 
-LIBS = @LIBS@ -lc
-
 AM_CFLAGS = @TCL_CFLAGS@
 
 man = gdtclft.3tcl
@@ -28,12 +26,12 @@ endif
 
 libgdtclft_la_SOURCES = gdtclft.c
 
-libgdtclft_la_LDFLAGS = -no-undefined
+libgdtclft_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if WITH_LIBGD
 libgdtclft_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	$(GD_LIBS)
 
 if WITH_TCL
--- origsrc/graphviz-2.36.0/tclpkg/gv/Makefile.am	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/tclpkg/gv/Makefile.am	2014-02-09 21:01:11.897004900 -0600
@@ -13,8 +13,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/cdt \
 	-I$(top_srcdir)/lib/pathplan
 
-LIBS = -lc
-
 BASESOURCES = gv.cpp gv.i gv_builtins.c
 BASELIBS = $(top_builddir)/lib/gvc/libgvc.la \
 	$(top_builddir)/lib/cgraph/libcgraph.la \
@@ -260,9 +258,9 @@ pkgtcldir = $(pkglibdir)/tcl
 TCL_data = 
 nodist_libgv_tcl_la_SOURCES = gv_tcl.cpp $(TCL_data)
 libgv_tcl_la_SOURCES = $(BASESOURCES) gv_tcl_init.c
-libgv_tcl_la_LIBADD = $(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la $(BASELIBS) $(TCL_LIBS)
+libgv_tcl_la_LIBADD = $(TCL_STUB_LIB_SPEC) $(BASELIBS) $(TCL_LIBS)
 libgv_tcl_la_LDFLAGS = -module -avoid-version
-libgv_tcl_la_CPPFLAGS = $(BASECPPFLAGS) $(TCL_INCLUDES)
+libgv_tcl_la_CPPFLAGS = $(BASECPPFLAGS) $(TCL_CFLAGS) $(TCL_INCLUDES)
 # $(TCL_data): gv_tcl.cpp
 gv_tcl.cpp: gv.i
 	$(SWIG) -c++ -tcl -namespace -o gv_tcl.cpp $(srcdir)/gv.i
--- origsrc/graphviz-2.36.0/tclpkg/tcldot/Makefile.am	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/tclpkg/tcldot/Makefile.am	2014-02-09 21:01:11.901005100 -0600
@@ -20,8 +20,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/cdt \
 	-I$(top_srcdir)/lib/pathplan $(TCLHANDLE_INCLUDES) $(GD_INCLUDES) $(TCL_INCLUDES)
 
-LIBS = -lc
-
 AM_CFLAGS = $(TCL_CFLAGS)
 
 man = tcldot.3tcl
@@ -51,10 +49,10 @@ libtcldot_la_SOURCES = \
 	tcldot-io.c \
 	no_builtins.c $(GDTCLFT)
 libtcldot_la_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
-libtcldot_la_LDFLAGS = -no-undefined 
+libtcldot_la_LDFLAGS = -module -avoid-version -no-undefined 
 
 libtcldot_la_LIBADD = $(TCLHANDLE_LIBS) \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	$(top_builddir)/lib/cdt/libcdt.la \
 	$(top_builddir)/lib/cgraph/libcgraph.la \
 	$(top_builddir)/lib/pathplan/libpathplan_C.la \
@@ -79,7 +77,7 @@ libtcldot_builtin_la_SOURCES = \
 	tcldot-io.c \
 	tcldot_builtins.c $(GDTCLFT)
 libtcldot_builtin_la_CPPFLAGS = $(AM_CPPFLAGS) -DDEMAND_LOADING=1
-libtcldot_builtin_la_LDFLAGS =
+libtcldot_builtin_la_LDFLAGS = -module -avoid-version -no-undefined
 
 libtcldot_builtin_la_LIBADD = $(TCLHANDLE_LIBS) \
 	$(top_builddir)/plugin/dot_layout/libgvplugin_dot_layout_C.la \
@@ -89,7 +87,7 @@ libtcldot_builtin_la_LIBADD = $(TCLHANDL
 	$(top_builddir)/lib/cgraph/libcgraph_C.la \
 	$(top_builddir)/lib/cdt/libcdt_C.la \
 	$(top_builddir)/lib/pathplan/libpathplan_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la $(GTS_LIBS)
+	$(TCL_STUB_LIB_SPEC) $(GTS_LIBS)
 
 if WITH_PANGOCAIRO
 libtcldot_builtin_la_LIBADD += $(top_builddir)/plugin/pango/libgvplugin_pango_C.la $(PANGOCAIRO_LIBS) $(PANGOFT2_LIBS)
--- origsrc/graphviz-2.36.0/tclpkg/tclpathplan/Makefile.am	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/tclpkg/tclpathplan/Makefile.am	2014-02-09 21:01:11.904005300 -0600
@@ -18,10 +18,10 @@ pkgtcl_LTLIBRARIES = libtclplan.la
 endif
 
 libtclplan_la_SOURCES = find_ints.c intersect.c makecw.c tclpathplan.c wrapper.c
-libtclplan_la_LDFLAGS = -no-undefined
+libtclplan_la_LDFLAGS = -module -avoid-version -no-undefined
 libtclplan_la_LIBADD = \
 	$(top_builddir)/tclpkg/tclhandle/libtclhandle_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la \
+	$(TCL_STUB_LIB_SPEC) \
 	$(top_builddir)/lib/pathplan/libpathplan.la $(MATH_LIBS)
 
 if WITH_TCL
--- origsrc/graphviz-2.36.0/tclpkg/tkspline/Makefile.am	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/tclpkg/tkspline/Makefile.am	2014-02-09 21:01:11.906005400 -0600
@@ -26,11 +26,10 @@ man_MANS = $(man)
 pdf_DATA = $(pdf)
 endif
 
-libtkspline_la_LDFLAGS = -no-undefined 
+libtkspline_la_LDFLAGS = -module -avoid-version -no-undefined 
 libtkspline_la_SOURCES = tkspline.c
 libtkspline_la_LIBADD = \
-	$(top_builddir)/tclpkg/tkstubs/libtkstubs_C.la \
-	$(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la @TK_LIB_SPEC@
+	$(TK_STUB_LIB_SPEC) $(TCL_STUB_LIB_SPEC)
 
 if WITH_TK
 all-local: pkgIndex.tcl
