--- origsrc/graphviz-2.47.0/configure.ac	2021-03-16 09:04:01.000000000 +0900
+++ src/graphviz-2.47.0/configure.ac
@@ -101,7 +101,7 @@
 GRAPHVIZ_VERSION_BUILD=0
 GRAPHVIZ_VERSION_REVISION=0
 case "${host_os}" in
-  *uwin* | *cygwin* | *mingw32* )
+  *uwin* | *mingw32* )
 	# Windows version numbers
 	# For the build number: months since Jan 2000, day of month from the timestamp
 	# For the revision number: hour, minute from the timestamp
@@ -154,8 +154,8 @@
 	;;
   *cygwin* )
 	CYGWIN=yes
-	DEFAULT_FONTPATH="C:/WINDOWS/FONTS;C:/WINNT/Fonts;C:/winnt/fonts"
-	PATHSEPARATOR=";"
+	DEFAULT_FONTPATH="/cygdrive/c/Windows/Fonts:/usr/share/X11/fonts/Type1:/usr/share/X11/fonts/TTF"
+	PATHSEPARATOR=":"
 	BROWSER="xdg-open"
 	JSHEXT=so
 	;;
@@ -202,7 +202,7 @@
 AC_SUBST([JSHEXT])
 AC_DEFINE_UNQUOTED(DEFAULT_FONTPATH,"$DEFAULT_FONTPATH",[Path to TrueType fonts.])
 AC_DEFINE_UNQUOTED(PATHSEPARATOR,"$PATHSEPARATOR",[Path separator character.])
-AM_CONDITIONAL(WITH_WIN32, [test "x$UWIN" = "xyes" -o "x$CYGWIN" = "xyes" -o "x$MINGW32" = "xyes"])
+AM_CONDITIONAL(WITH_WIN32, [test "x$UWIN" = "xyes" -o "x$MINGW32" = "xyes"])
 AM_CONDITIONAL(WITH_DARWIN9, [test "x$DARWIN9" = "xyes"])
 
 DEFAULT_DPI=96
@@ -253,7 +253,7 @@
 	AC_ENABLE_SHARED
         use_shared="Yes"
 	case "${host_os}" in
-  		*uwin* | *cygwin* | *mingw32* )
+  		*uwin* | *mingw32* )
 			AC_DEFINE_UNQUOTED(GVDLL,1, [Define for DLLs on Windows.])
 		;;
 	esac
